include/dem/read_checkpoint.h

```diff
@@ -41,6 +41,7 @@ read_checkpoint(
   parallel::distributed::Triangulation<dim>               &triangulation,
   Particles::ParticleHandler<dim>                         &particle_handler,
   std::shared_ptr<Insertion<dim>>                         &insertion_object,
-  std::vector<std::shared_ptr<SerialSolid<dim - 1, dim>>> &solid_surfaces);
+  std::vector<std::shared_ptr<SerialSolid<dim - 1, dim>>> &solid_surfaces,
+  CheckpointControl &checkpoint_controller);
```

Comment:
We need to pass the checkpoint controller as a reference in the read and write since we increment the <next_checkpoint_id> in both function. (Tell me if I'm wrong)

Reply:
Yeah, that's true, or you could just pass the checkpoint id and let the checkpoint controller increment.

Reply:
With this new implementation, it is now const. Is it ok?

Reply:
Yeah if it's const now it's fine